sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/mdesupport/EncryptionKeyWrapProvider.java

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.encryption.mdesupport;
+
+import com.microsoft.data.encryption.cryptography.MicrosoftDataEncryptionException;
+
+public abstract class EncryptionKeyWrapProvider {
```

Comment:
Why do we need abstract class here? Can't we make this as an interface?

Reply:
Once AAP java support catching , we can provide default implementation here and give user the way to override same as .NET https://github.com/Azure/azure-cosmos-dotnet-v3/blob/5c01524de44ea0104223f6d494edeb98ac30a44c/Microsoft.Azure.Cosmos.Encryption/src/MdeSupport/EncryptionKeyWrapProvider.cs#L32